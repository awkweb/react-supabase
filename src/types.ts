import { PostgrestFilterBuilder } from '@supabase/postgrest-js'

export type Filter<Data> = (
    query: PostgrestFilterBuilder<Data>,
) => PostgrestFilterBuilder<Data>

export type PostgrestError = {
    message: string
    details: string
    hint: string
    code: string
}
