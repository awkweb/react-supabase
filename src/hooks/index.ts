export * from './use-client'
export * from './use-filter'
export * from './use-select'
