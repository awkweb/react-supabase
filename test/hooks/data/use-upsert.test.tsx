import { renderHook } from '@testing-library/react-hooks'

import { useUpsert } from '../../../src'

describe('useUpsert', () => {
    it('should throw when not inside Provider', () => {
        const { result } = renderHook(() => useUpsert('todos'))
        expect(() => result.current).toThrowErrorMatchingSnapshot()
    })
})
