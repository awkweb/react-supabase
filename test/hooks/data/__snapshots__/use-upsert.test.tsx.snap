// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useUpsert should throw when not inside Provider 1`] = `"No client has been specified using Provider."`;
